src/Surface.zig

```diff
@@ -646,6 +646,16 @@ pub fn init(
             // an initial size shouldn't stop our terminal from working.
             log.warn("unable to set initial window size: {s}", .{err});
         };
+
+        if (config.@"window-maximize") {
+            rt_app.performAction(
+                .{ .surface = self },
+                .toggle_maximize,
+                {},
+            ) catch |err| {
+                log.warn("unable to maximize window: {s}", .{err});
+            };
+        }
```

Comment:
This should not happen in `src/Surface.zig`. Every new tab or split you create would maximize the window even if the user had manually unmaximized the window.

Maximizing the window when it's created should happen here:

https://github.com/ghostty-org/ghostty/blob/918ccdba5cc65ccd1fb48a54c71306d869299441/src/apprt/gtk/Window.zig#L268-L269